contracts/ProducerContract.sol

```diff
@@ -153,7 +154,7 @@ contract ProducerContract is Callable {
     if (producer.isa.sustainable) return true;
     if (newProducerScore < 0) isaScore = isaScore - (newProducerScore);
 
-    producersTotalScore += isaScore;
+    if (producer.syntropicProducer) producersTotalScore += isaScore;
```

Comment:
Essa linha aqui faz com que `producersTotalScore` quando a condição for verdadeira, no entanto o `producer.isa.isaScore` é atualizado independente dessa condição.


A forma como chamamos o `pool` é afetada por isso.

Quando chamamos o `pool`  hoje nós enviamos `producersTotalScore` e `producer.isa.isaScore`.
```solidity
    producerPool.withdraw(
      msg.sender,
      producersTotalScore,
      producer.isa.isaScore,
      producer.pool.currentEra
    );
```

<hr>

### fazendo alguns testes (Supondo que temos apenas um produtor)

Quando o produtor tem a primeira inspeção realizada e `recebe 70 de isaScore`
Não consegue sacar ainda
```solidity
producersTotalScore == 0
producer.isa.isaScore == 70
producer.totalInspections == 1
```
<br>
<br>

Quando o produtor tem a segunda inspeção realizada e `recebe 100 de isaScore`
Não consegue sacar ainda
```solidity
producersTotalScore == 0
producer.isa.isaScore == 170
producer.totalInspections == 2
```

<br>
<br>

Quando o produtor tem a terceira inspeção realizada e `recebe 30 de isaScore`
Consegue sacar na próxima era
```solidity
producersTotalScore == 0
producer.isa.isaScore == 200
producer.totalInspections == 3
```


Nesse cenário quando o produtor for tentar sacar o `producersTotalScore` vai está zerado, mas o `producer.isa.isaScore` vai ter valor. Nai vai conseguir sacar nada.

No entanto quando quando ele ter uma outra inspeção realizada e receber +50 de isaScore então ficaria assim
```solidity
producersTotalScore == 50
producer.isa.isaScore == 250
producer.totalInspections == 4
```

e quando ele solicitasse o saque (vamos supor que o pool distribui 50.000 por era)
(50.000 / 50) * 250 = 250000 (Iria conseguir sacar 250 mil tokens sendo que o máximo era 50.000).


Uma solução pra isso seria criar um novo state de isaScores do produtor que não vão pra a pool

Reply:
Entendi o problema.

Consegue explicar melhor essa parte de criar um novo state de isaScores?

Reply:
Seria um state só pra guardar o isaScore que ele recebeu antes de entrar na pool

Reply:
Não somaria ao producer.isa.isaScore, pois esse é enviado pra a pool